src/components/Toolbox.tsx

```diff
@@ -82,11 +101,41 @@ export default function Toolbox(props: React.PropsWithChildren<ToolboxProps>) {
             {/* Zoom Out Button */}
             <ActionButton onClick={StateManager.zoomOut} icon={<BsZoomOut />} title="Zoom Out" bgColor="bg-blue-500"></ActionButton>
             {/* Clear Stage No Save Button */}
-            <ActionButton onClick={StateManager.clearMachine} icon={<BiFileBlank />} title="New Automaton" bgColor="bg-black" margin="m-10"></ActionButton>
+            <div className="flex flex-col items-center mt-4"></div>
+            <ActionButton onClick={handleClearMachineClick}icon={<BiFileBlank />}title="New Automaton"bgColor="bg-black" margin="m-10"></ActionButton>
+      
+      
             {/* Undo Button */}
             <ActionButton onClick={StateManager.undoState} icon={<BsFillArrowLeftCircleFill />} title="Undo most recent action" bgColor="bg-blue-500"></ActionButton>
             {/* Redo Button */}
             <ActionButton onClick={StateManager.redoState} icon={<BsFillArrowRightCircleFill />} title="Redo most recent action" bgColor="bg-blue-500"></ActionButton>                               
         </div>
+        {/*  This does not work as intended. It will clear the screen as it should but the animations and closable window are broken and causes lag.
```

Comment:
For merging into the main code, I've removed this comment block since it doesn't do anything